### ISSUE TITLE
Ability to copy cover art from clipboard

### ISSUE THREAD

**Issue by Walrus94** [keithgg/puddletag#428]
Hi!
In some cases it would be very handful to use clipboard instead of file manager to add cover art image (for instance from web browser).

Thanks!

**Comment by twiztedtbone** [comment]
agreed, this was easily one of my most used features in mp3tag. while it _appears_ to be implemented in puddletag, it doesn't actually work. I can copy a cover art from e.g. google images and paste it into the artwork window, but it won't stay in the tag no matter how many times I click save, nor does puddletag error/crash or anything, it just does nothing :(

**Comment by pedrobaeza** [comment]
I added this support in #652, so this can be closed. Although I have cloned latest source code, and it seems to be broken. @corubba maybe one of your refactorings?

**Comment by corubba** [comment]
Whatever the reason, if it's broken we fix it. I will keep this open but turn it from a feature request into a bug to keep track of it.

**Comment by pedrobaeza** [comment]
Thanks! Let me know any doubt, or next week I can work on it.

**Comment by pedrobaeza** [comment]
The specific problem is that you paste it, and that works, but it's not saved. Maybe the problem is the same if you add regularly the cover.

**Comment by corubba** [comment]
So that copy/paste is only (properly) implemented in the "Artwork window", not in "Extended Tags" or the "Full-Size Artwork View" window (when you left-click on artwork). And it is *mostly* working for me with the current `master` branch.
I can copy the currently shown artwork to the clipboard, paste new artwork in and hit "Save" to persist it. Also tested different formats like mp3 and ogg, all work. Pasting artwork will not replace any existing artwork, it will be appended. And hitting "Save" will reset the artwork window to always show the first image.
What is not working properly is pasting in "Preview Mode". You can paste an image, it shows up in the artwork window. But it is not registered as a change, that file will not be marked with the green background in the maintable. Switch to another file and back, and the image is gone from the artwork window. If you hit "Write Previews" directly after pasting, it won't write anything and also not disable "Preview Mode". The same happens with "Add Cover" and "Remove Cover", this looks like a general issue with the interactions in the "Artwork window" during "Preview Mode".
